sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -529,12 +533,10 @@ rerouteOption
     ;
 
 dataType
-    : IDENTIFIER
-    | quotedIdentifier
-    | nonReserved
-    | objectTypeDefinition
-    | arrayTypeDefinition
-    | setTypeDefinition
+    : ident                     #dataTypeIdent
```

Comment:
replacing `quotedIdentifier`, `nonReserved`, `IDENTIFIER` with ident results in exactly the same behavior.